src/arrayMethodSort.js

```diff
@@ -5,7 +5,24 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    const defaultCompare = (a, b) => {
+      const first = a.toString();
+      const second = b.toString();
+
+      return first > second ? 1 : (first < second ? -1 : 0);
+    };
+
+    const compare = compareFunction || defaultCompare;
+
+    for (let i = 0; i < this.length; i++) {
+      for (let j = 0; j < (this.length - i - 1); j++) {
+        if (compare(this[j], this[j + 1]) > 0) {
+          [this[j], this[j + 1]] = [this[j + 1], this[j]];
+        }
+      }
+    }
```

Comment:
Use `while` construction instead of first `for` loop to describe the conditions under which the program will run.